pkg/tests/apis/iam/iam_test.go

```diff
@@ -173,3 +176,142 @@ func TestIntegrationIdentity(t *testing.T) {
 		]`, found)
 	})
 }
+
+func TestIntegrationUsers(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	t.Run("User CRUD operations with dual writer mode 0", func(t *testing.T) {
+		helper := apis.NewK8sTestHelper(t, testinfra.GrafanaOpts{
+			AppModeProduction:    false,
+			DisableAnonymous:     true,
+			APIServerStorageType: "unified",
+			UnifiedStorageConfig: map[string]setting.UnifiedStorageConfig{
+				"users.iam.grafana.app": {
+					DualWriterMode: grafanarest.Mode0,
+				},
+			},
+			EnableFeatureToggles: []string{
+				featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs,
+			},
+		})
+		doUserCRUDTests(t, helper)
+	})
+
+	t.Run("User CRUD operations with dual writer mode 1", func(t *testing.T) {
+		helper := apis.NewK8sTestHelper(t, testinfra.GrafanaOpts{
+			AppModeProduction:    false,
+			DisableAnonymous:     true,
+			APIServerStorageType: "unified",
+			UnifiedStorageConfig: map[string]setting.UnifiedStorageConfig{
+				"users.iam.grafana.app": {
+					DualWriterMode: grafanarest.Mode1,
+				},
+			},
+			EnableFeatureToggles: []string{
+				featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs,
+			},
+		})
+		doUserCRUDTests(t, helper)
+	})
+}
+
+func doUserCRUDTests(t *testing.T, helper *apis.K8sTestHelper) {
+	t.Run("should create user and delete it using the new APIs", func(t *testing.T) {
+		ctx := context.Background()
+		userClient := helper.GetResourceClient(apis.ResourceClientArgs{
+			User: helper.Org1.Admin,
```

Comment:
Could you also add a check that only admins can do this? (or whichever users are supposed to be able to?)